journals/apps/journals/models.py

```diff
@@ -208,6 +220,32 @@ def create_journal_access(cls, user, journal, order_number=None):
         access.save()
         return access
 
+    @classmethod
+    def bulk_create_journal_access(cls, usernames, journal_id, expiration_date=None):
+        """
+        Bulk create the JournalAccess on the given parameters
+
+        Args:
+            usernames (set): Set of valid usernames.
+            journal_id: Journal's id on which we want to give access to users.
+            expiration_date:  Journal access' expiration date for all the users.
+        """
+        journal_access_list = []
+        journal = Journal.get_journal_by_id(journal_id)
+        if not expiration_date:
+            expiration_date = datetime.datetime.now() + datetime.timedelta(days=journal.access_length)
+        for username in usernames:
+            user = User.get_user_by_username(username)
+            if user:
```

Comment:
In the rare case that a user gets passed in the valid_usernames, and then is None here, we should log it.